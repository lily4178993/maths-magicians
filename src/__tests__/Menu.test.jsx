import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import Menu from '../components/Menu';

describe('Menu Component', () => {
  it('matches snapshot', () => {
    const tree = renderer
      .create(
        <Router>
          <Menu />
        </Router>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders Home link', () => {
    render(
      <Router>
        <Menu />
      </Router>,
    );

    const homeLink = screen.getByText('Home');
    expect(homeLink).toBeInTheDocument();
    expect(homeLink).toHaveAttribute('href', '/');
  });

  it('renders Calculator link', () => {
    render(
      <Router>
        <Menu />
      </Router>,
    );

    const calculatorLink = screen.getByText('Calculator');
    expect(calculatorLink).toBeInTheDocument();
    expect(calculatorLink).toHaveAttribute('href', '/calculator');
  });

  it('renders Quote link', () => {
    render(
      <Router>
        <Menu />
      </Router>,
    );

    const quoteLink = screen.getByText('Quote');
    expect(quoteLink).toBeInTheDocument();
    expect(quoteLink).toHaveAttribute('href', '/quote');
  });
});
