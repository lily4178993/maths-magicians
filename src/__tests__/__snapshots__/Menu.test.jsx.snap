// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu Component matches snapshot 1`] = `
<nav>
  <ul>
    <li>
      <a
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <li>
      <a
        href="/calculator"
        onClick={[Function]}
      >
        Calculator
      </a>
    </li>
    <li>
      <a
        href="/quote"
        onClick={[Function]}
      >
        Quote
      </a>
    </li>
  </ul>
</nav>
`;
